lib/main.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |x|
+      new_arr << x if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all(arg = nil)
+    unless block_given?
+      if arg.nil?
+        falsy = 0
+        my_each do |x|
+          falsy += 1 if x == false
+        end
+        return falsy.zero?
+      end
+    end
```

Comment:
Please, it's not working. There is a bug
```ruby
irb(main):010:0> [nil, true, 99].all?
=> false
irb(main):011:0> [nil, true, 99].my_all
=> true
```

Reply:
Thanks. This is resolved.